tests/util/caches/test_descriptors.py

```diff
@@ -703,6 +704,48 @@ async def list_fn(self, args1, arg2):
             obj.mock.assert_called_once_with((40,), 2)
             self.assertEqual(r, {10: "fish", 40: "gravy"})
 
+    def test_concurrent_lookups(self):
+        """All concurrent lookups should get the same result"""
+
+        class Cls:
+            def __init__(self):
+                self.mock = mock.Mock()
+
+            @descriptors.cached()
+            def fn(self, arg1):
+                pass
```

Comment:
I don't think this is used---or is the `"fn"` on 718 referring to this somehow?

Reply:
> is the `"fn"` on 718 referring to this somehow?

yes, it is. an `@cachedList` function is alwaysan extension of an `@cached` function - it doesn't actually have a cache of its own.

So if you don't really care about having a non-list version, you still have to have a stub, just to hold the cache.

There are a few instances of this in the codebase, though there they throw `NotImplementedError`.